src/test/java/wolox/training/controllers/UserControllerTest.java

```diff
@@ -0,0 +1,284 @@
+
+        String json = mapper.writeValueAsString(userTest);
+
+        // When + Then
+
+        mvc.perform(MockMvcRequestBuilders.put("/api/users/1")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(json)
+                .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+        System.out.println("whenUpdateAnUser Passed");
+    }
+
+    @Test
+    void whenUpdateAnUserIdMismatch_thenThrowIdMismatched() throws Exception {
+        //Given
+        User userTest = new User(1L,"Username","David", LocalDate.parse("2004-09-25"),new ArrayList<>());
+
+        given(userRepository.save(userTest)).willReturn(userTest);
+
+        userTest.setName("Santiago");
+        given(userRepository.save(userTest)).willReturn(userTest);
+        given(userRepository.findById(userTest.getId())).willReturn(java.util.Optional.of(userTest));
+
+        String json = mapper.writeValueAsString(userTest);
+
+        // When + Then
+
+        mvc.perform(MockMvcRequestBuilders.put("/api/users/2")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(json)
+                .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isBadRequest());
+        System.out.println("whenUpdateAnUserIdMismatch Passed");
+    }
+
+    @Test
+    void whenDeleteAnUserById_thenReturnString() throws Exception {
+        //Given
+        User userTest = new User(1L,"Username","David", LocalDate.parse("2004-09-25"),new ArrayList<>());
+
+        given(userRepository.findById(userTest.getId())).willReturn(java.util.Optional.of(userTest));
+
+        // When + Then
+        mvc.perform(MockMvcRequestBuilders.delete("/api/users/1")
+                .contentType(MediaType.APPLICATION_JSON)
+                .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+        System.out.println("whenDeleteAnUserById Passed");
+    }
+
+    @Test
+    void whenAddABookToAnUserLibrary_thenReturnOk() throws Exception {
+        //Given
+        User userTest = new User(1L,"Username","David", LocalDate.parse("2004-09-25"),new ArrayList<>());
+        given(userRepository.findById(userTest.getId())).willReturn(java.util.Optional.of(userTest));
+
+        Book bookTest = new Book(1L,"Mystery","Paolo", "image.png", "Sorcerer's Stone", "-", "Salamander","2000", 250,"123456789");
+        userTest.addBookToUser(bookTest);
+        given(userRepository.save(userTest)).willReturn(userTest);
+
+        String json = mapper.writeValueAsString(userTest);
+
+        // When + Then
+        mvc.perform(MockMvcRequestBuilders.post("/api/users/1/addBook")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(json)
+                .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+        System.out.println("whenAddABookToAnUserLibrary Passed");
```

Comment:
This print is not necessary, as it is a test, the result is already shown by the console